src/app/utils/BlueprintFunctions.js

```diff
@@ -327,26 +327,43 @@ export function setPercent(ship, m, percent) {
   ship.clearModifications(m);
   // Pick given value as multiplier
   const mult = percent / 100;
-  const features = m.blueprint.grades[m.blueprint.grade].features;
+  setQualityCB(m.blueprint, mult, (featureName, value) => ship.setModification(m, featureName, value));
+}
+
+/**
+ * Sets the blueprint quality and fires a callback for each property affected.
+ * @param {Object}   blueprint  The ship for which to perform the modifications
+ * @param {Number}   quality    The quality to apply - float number 0 to 1.
+ * @param {Function} cb         The Callback to run for each property. Function (featureName, value)
+ */
+export function setQualityCB(blueprint, quality, cb) {

```

Comment:
This call back is called for each modified module property, since one blueprint has multiple properties and thus the call back will be fired multiple times.
Alternatively I can change this function to return a set of properties and then they can be cycled from where the function is called.

Reply:
Not necessary. In my mental model, callbacks are only used for asynchronous behavior. But you're right, e.g., `forEach` also takes callback.